### ISSUE TITLE
YOLO-NAS model loss turning to nan's during training

### ISSUE THREAD

**Issue by wematan** [Deci-AI/super-gradients#1081]
### 🐛 Describe the bug

Trying to train yolo-nas on a custom dataset, the training starts off quite well and it seems like learning is indeed happening, while the loss decreases from: 
**(iteration 0):**
``` 
Train epoch 0:   0% 10/99420 [00:50<29:36:51,  1.07s/it, PPYoloELoss/loss=9.62, PPYoloELoss/loss_cls=7.92, PPYoloELoss/loss_dfl=1.39, PPYoloELoss/loss_iou=0.401
```
to **(iteration 77284)**:
``` 
Train epoch 0:  78% 77284/99420 [78:25:29<15:56:01,  2.59s/it, PPYoloELoss/loss=1.59, PPYoloELoss/loss_cls=0.757, PPYoloELoss/loss_dfl=0.738, PPYoloELoss/loss_iou=0.185, gpu_mem=9.02] 
```

but just after the 77284 iteration the model starts to produce nan losses.
```
Train epoch 0:  78% 77291/99420 [78:25:34<6:46:42,  1.10s/it, PPYoloELoss/loss=nan, PPYoloELoss/loss_cls=nan, PPYoloELoss/loss_dfl=nan, PPYoloELoss/loss_iou=nan, gpu_mem=9.02]
```

I tried to train the model again as well, but while the input is good (verified the input image), when "forward"ing the first sample it produces numeric loss but any sample afterwards that is fed into the networks backbone produces 'nan's.

![image](https://github.com/Deci-AI/super-gradients/assets/5028096/afc2d0e7-afd1-4c5b-87dd-79ae4b0e15ce)
As you can see the model's backbone produces all nan's. (# Blurred the image due to privacy)

I've also tried to inspect the model weights and didn't find any issues, no zero/inf/nan parameters. 

after further debugging it seems that the problem resides in AMP, when checking the scale (which is multiplied by loss) is zero, when i return turns the loss to zero.

### Versions

Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.25.0
Libc version: glibc-2.31

Python version: 3.8.10 (default, Mar 13 2023, 10:26:41)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-69-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3050
Nvidia driver version: 525.105.17
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.4
[pip3] torch==1.13.0+cu117
[pip3] torchaudio==0.13.0+cu117
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.14.0+cu117

**Comment by BloodAxe** [comment]
What batch size are you using for training? I see that you have 3050, so I assume it has 8Gb of VRAM. 
If you decreased batch size by a factor of two (for example) then you also want to reduce learning rate at the same scale. This may help to address the NaNs during training. Second option is to disable mixed precision, but this would effectively double the VRAM requirements for training. 
I suggest you to try with lower LR or smaller model (since you will be able to fit bigger batch) to stabilize the training.

**Comment by wematan** [comment]
Hi, 
I was training on a dedicated machine with 8XA4000 (16GB). im just debugging on my dev PC (with RTX3050).
I tried lowering the learning rate, it did enabled a few more iterations before converging to nan but obviously didn't work.
Disabling mixed precision worked but results in a significant increase in training time.

BR,
Matan.

**Comment by BloodAxe** [comment]
It is hard to say why you have issues with fine-tuning on your dataset.
We've been successfully able to fine tune Yolo NAS on 100 diverse datasets (RF-100) using following recipes:

* [Yolo-NAS S](https://github.com/Deci-AI/super-gradients/blob/4c2f8c31ac696664efd3809a7896c3f1472217ad/src/super_gradients/recipes/roboflow_yolo_nas_s.yaml)
* [Yolo-NAS M](https://github.com/Deci-AI/super-gradients/blob/4c2f8c31ac696664efd3809a7896c3f1472217ad/src/super_gradients/recipes/roboflow_yolo_nas_m.yaml)

I suggest that you try with these parameters. If the problem persist, the most likely reason is the data itself. On possible case - big number of empty batches (W/o bboxes).

**Comment by BloodAxe** [comment]
Closing as stale
